### PR TITLE
Setup of redux

### PR DESCRIPTION
**Pull request by Hassaanjbaig-code**
## In this pull request 

- Setup of redux 
- Add a book reduced with the action of **AddBooks** and **RemoveBooks**
- Add a categories reducer with the action of **CheckStatus**
- Setup of configuration

### CODE REVIEW

File: src/Component/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const RemoveBooks = createAction('books/remove');
+
+const books = createReducer(initialstate, (builder) => {
+  builder.addCase(AddBooks, (state, action) => {
+    state.push(action.payload);
+  });
+  builder.addCase(RemoveBooks, (state, action) => {
+    state.filter((book) => book.id !== action.payload);
+  });
+  builder.addDefaultCase((state) => state);
```
Comment (emmiiorji) [inline src/Component/redux/books/books.js]: Your reducers have been well implemented. However, following the ducks pattern as [recommended](https://github.com/microverseinc/curriculum-react-redux/blob/ea3be79fa8aeba02a7d576dac93fcf0a684e16c5/bookstore/add_reducers_an_actions.md?plain=1#L29), kindly ensure that the redux directory is directly within the `src` directory and similar to the illustration below:

image

File: src/Component/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  {
+    id: 3,
+    bookcategory: 'Economy',
+    Title: 'Capital in the Twenty-First Century',
+    Author: 'Suzanne Collins',
+  },
+];
+
+const AddBooks = createAction('books/add');
+const RemoveBooks = createAction('books/remove');
```
Comment (emmiiorji) [inline src/Component/redux/books/books.js]: Kindly ensure that the action types follow the ducks pattern. e.g `bookstore/books/REMOVE_BOOK`

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @Hassaanjbaig-code, 
# Status: Changes Required ♻️

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

A little bit more

## Good Highlights✔️

- Linter checks are passing
- Good commit messages
- Proper use of Gitflow


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Hassaanjbaig-code** [comment]
@emmiiorji  Thank you for your review
I will do and submit it again

**Review by emmiiorji — APPROVED** [review]
Hi @Hassaanjbaig-code, 

# Status: Approved ✅

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Congrats

## Good Highlights✔️

- Requested changes have been implemented.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Hassaanjbaig-code** [comment]
@emmiiorji **Thank you for your review** 
and approving the project.

**Pull request merged**